docs/fundamentals/gateway/custom-gateway.md

```diff
@@ -0,0 +1,324 @@
+(custom-gateway)=
+# Customize the Gateway
+Jina Gateways are customizable, and you can implement them in much the same way as an Executor.
+With Custom Gateways, Jina gives power back to the users, by allowing them to implement any server, protocol and 
+interface at the gateway level. This means you have more freedom to:
+* Define and expose your own API Gateway interface to clients. You can define your JSON schema or protos,...
+* Choose your favourite server framework.
+* Choose the protocol used to serve your app.
+
+Customization is allowed different components:
+* Implement the custom gateway using a `base` gateway class: {class}`~jina.Gateway` or {class}`~jina.serve.runtimes.gateway.http.fastapi.FastAPIBaseGateway`.
+* Use the {class}`~jina.serve.streamer.GatewayStreamer` to send data to Executors in the Flow.
+* Implement the needed health-checks for Jina.
+* Optionally, define a `config.yml` for it.
+* Optionally, bootstrap the gateway using a `Dockerfile` and re-using the docker image.
```

Comment:
I would not mention these things here at the beginning. I think it is nicer if we keep things high-level, so that the user knows what is even going on, and then we go into increasingly more detail once these things become actually relevant to the implementation.

I would keep the message around "Custom Gateway works like Executor". Then, if they are already familiar, they can already imagine the support for yaml, docker, ... And if they are not already familiar, then telling them all of this is too much info anyways imo

Reply:
I removed these lines and spread them out (as intros/transitions in subsequent sections)